Dockerfile

```diff
@@ -0,0 +1,13 @@
+FROM node:10 as build-stage
+
+RUN mkdir /usr/src/app
+WORKDIR /usr/src/app
+
+ARG API_URL
+ENV API_URL=${API_URL}
```

Comment:
With a CI provider like Buildkite or Jenkins, they'll usually offer you a way to place environment variables like those in your local .env files in the machine environment steps so you wouldn't need to define ARG and ENV variables. I would put in the docker-compose environment property only the environment variables you will be using to help form anything dynamic i.e. commands, images names, etc. For us we would source a config file like `source conf/testing.conf` to hydrate our environment with the proper variables, which does a bunch of `export SOME_ENV_VARIABLE=<SOME_ENV_VALUE>` and then run the build command.